### DISCUSSION TITLE
Please add an event in ElementQuery::afterPopulate

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#11262] [Ideas] [answered]
I often have custom field values that should be eager loaded that are no elements. Most of the time I extend your Element classes as well as your ElementQuery classes in order to access the [afterPopulate](https://github.com/craftcms/cms/blob/4.1/src/elements/db/ElementQuery.php#L1403) function so I can do custom eager loading like [Commerce Orders](https://github.com/craftcms/commerce/blob/develop/src/elements/db/OrderQuery.php#L1150-L1177) do,

It would be awesome if there was an event for this so I could do custom eager loading via behavior instead of extending the class.

So basically just
```PHP
/**
 * @inheritdoc
 */
public function afterPopulate(array $elements): array
{
    $event = new SomeElementEventName(['elements' => $elements]);
    if($this->hasEventHandlers(self::EVENT_AFTER_POPULATE_ELEMENTS)){
        $this->trigger(self::EVENT_AFTER_POPULATE_ELEMENTS, $event);
    }
    
    return $event->elements;
}
```

**Comment by brandonkelly** [comment, marked as answer]
Added for 4.1 🎉  (41ba108c2c527a0b1a126670b07733e08d494359)

You can move to the `4.1` branch by changing your `craftcms/cms` requirement to `4.1.x-dev` in composer.json.

**Reply by Anubarak** [reply]
As always thank you so much for this.
You implementations of my feature requests are always highly appreciated.
Those things improve the way I can work with Craft CMS greatly.

**Reply by brandonkelly** [reply]
Craft 4.1 is out with this!
